src/anomalib/models/components/base/buffer_list.py

```diff
@@ -0,0 +1,108 @@
+"""Buffer List Mixin."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import torch
+from torch import nn
+
+
+class BufferListMixin(nn.Module):
+    """Buffer List Mixin.
+
+    This mixin is used to allow registering a list of tensors as buffers in a pytorch module.
+    """
+
+    def register_bufferlist(self, name: str, values: list[torch.Tensor], persistent: bool = True, **kwargs) -> None:
+        """Register a list of tensors as buffers in a pytorch module.
+
+        Each tensor is registered as a buffer with the name `_name_i` where `i` is the index of the tensor in the list.
+        To update and retrieve the list of tensors, we dynamically assign a descriptor attribute to the class.
+
+        Args:
+            name (str): Name of the bufferlist.
+            values (list[torch.Tensor]): List of tensors to register as buffers.
+            persistent (bool, optional): Whether the buffers should be saved as part of the module state_dict.
+                Defaults to True.
+            **kwargs: Additional keyword arguments to pass to `torch.nn.Module.register_buffer`.
+
+        Example:
```

Comment:
Would it be an idea to have this example in the main class docstring. User would see this example at the top of the documentation. Otherwise, they might need to scroll?

This is an idea for any other module in the library actually?